app/models/manageiq/providers/google/cloud_manager/event_catcher/event_producer.rb

```diff
@@ -0,0 +1,62 @@
+# Helper class responsible for polling a Google Pubsub topic and retreiving
+# event type messages. Parsing of the message is deferred to the caller of
+# #poll.
+class ManageIQ::Providers::Google::CloudManager::EventCatcher::EventProducer
```

Comment:
I think the patterned name is `Stream`, and there is a method called `each_event` or something that yields individual events.  @durandom Can you help out here?